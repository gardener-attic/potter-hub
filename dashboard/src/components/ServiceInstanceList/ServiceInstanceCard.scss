.Provisioned {
  background-color: #15cb2d;
}

.Failed {
  background-color: #cb1515;
}
