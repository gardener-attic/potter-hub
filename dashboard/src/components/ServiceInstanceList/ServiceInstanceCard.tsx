import * as React from "react";

import "../Catalog/CatalogItem.css";
import InfoCard from "../InfoCard";
import "./ServiceInstanceCard.css";

export interface IServiceInstanceCardProps {
  name: string;
  namespace: string;
  servicePlanName: string;
  serviceClassName: string;
  statusReason: string | undefined;
  link?: string;
  icon?: string;
}

function generalizeStatus(status: string) {
  if (status.match(/Provisioned|Success/)) {
    return "Provisioned";
  }
  if (status.match(/Failed|Error/)) {
    return "Failed";
  }
  return status;
}

const ServiceInstanceCard: React.SFC<IServiceInstanceCardProps> = props => {
  const { name, namespace, link, icon, serviceClassName, servicePlanName, statusReason } = props;
  return (
    <InfoCard
      title={name}
      link={link}
      icon={icon}
      info={`${servicePlanName} ${serviceClassName}`}
      tag1Content={namespace}
      tag2Class={statusReason && generalizeStatus(statusReason)}
      tag2Content={statusReason && generalizeStatus(statusReason)}
    />
  );
};

export default ServiceInstanceCard;
