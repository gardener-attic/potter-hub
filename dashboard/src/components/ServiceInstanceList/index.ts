import ServiceInstanceList from "./ServiceInstanceList";

export default ServiceInstanceList;
