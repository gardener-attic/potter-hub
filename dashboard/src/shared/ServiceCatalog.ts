import { JSONSchema6 } from "json-schema";
import * as urls from "../shared/url";
import { axiosWithAuth } from "./AxiosInstance";
import { IClusterServiceClass } from "./ClusterServiceClass";
import { K8S_REVERSE_PROXY_URL } from "./Kube";
import { IServiceInstance } from "./ServiceInstance";
import { IK8sList, IStatus } from "./types";

export class ServiceCatalog {
  public static async getServiceClasses() {
    return this.getItems<IClusterServiceClass>("clusterserviceclasses");
  }

  public static async getServiceBrokers() {
    return this.getItems<IServiceBroker>("clusterservicebrokers");
  }

  public static async getServicePlans() {
    return this.getItems<IServicePlan>("clusterserviceplans");
  }

  public static async deprovisionInstance(instance: IServiceInstance) {
    const { data } = await axiosWithAuth.delete(
      `${K8S_REVERSE_PROXY_URL}${instance.metadata.selfLink}`,
    );
    return data;
  }

  public static async syncBroker(broker: IServiceBroker) {
    const { data } = await axiosWithAuth.patch<IStatus>(
      urls.api.clusterservicebrokers.sync(broker),
      {
        spec: {
          relistRequests: broker.spec.relistRequests + 1,
        },
      },
      {
        headers: { "Content-Type": "application/merge-patch+json" },
        validateStatus: statusCode => true,
      },
    );
    return data;
  }

  public static async isCatalogInstalled(): Promise<boolean> {
    try {
      const { status } = await axiosWithAuth.get(this.endpoint);
      return status === 200;
    } catch (err) {
      return false;
    }
  }

  public static async getItems<T>(resource: string, namespace?: string): Promise<T[]> {
    const response = await axiosWithAuth.get<IK8sList<T, {}>>(
      this.endpoint + (namespace ? `/namespaces/${namespace}` : "") + `/${resource}`,
    );
    const json = response.data;
    return json.items;
  }

  private static endpoint: string = `${K8S_REVERSE_PROXY_URL}/apis/servicecatalog.k8s.io/v1beta1`;
}
export interface IK8sApiListResponse<T> {
  kind: string;
  apiVersion: string;
  metadata: {
    selfLink: string;
    resourceVersion: string;
  };
  items: T[];
}

export interface ICondition {
  type: string;
  status: string;
  lastTransitionTime: string;
  reason: string;
  message: string;
}

export interface IServiceBroker {
  metadata: {
    name: string;
    selfLink: string;
    uid: string;
    resourceVersion: string;
    generation: number;
    creationTimestamp: string;
    finalizers: string[];
  };
  spec: {
    url: string;
    authInfo: any; // Look into
    relistBehavior: string;
    relistDuration: string;
    relistRequests: number;
  };
  status: {
    conditions: ICondition[];
    reconciledGeneration: number;
    lastCatalogRetrievalTime: string;
  };
}

export interface IServicePlan {
  metadata: {
    name: string;
    selfLink: string;
    uid: string;
    resourceVersion: string;
    creationTimestamp: string;
  };
  spec: {
    clusterServiceBrokerName: string;
    externalName: string;
    externalID: string;
    description: string;
    externalMetadata?: {
      displayName: string;
      bullets: string[];
    };
    instanceCreateParameterSchema?: JSONSchema6;
    serviceBindingCreateParameterSchema?: JSONSchema6;
    free: boolean;
    clusterServiceClassRef: {
      name: string;
    };
  };
  status: {
    removedFromBrokerCatalog: boolean;
  };
}
