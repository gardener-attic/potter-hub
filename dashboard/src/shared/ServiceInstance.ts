import { axiosWithAuth } from "./AxiosInstance";
import { K8S_REVERSE_PROXY_URL } from "./Kube";
import { ICondition, ServiceCatalog } from "./ServiceCatalog";
import { IStatus } from "./types";

export interface IServiceInstance {
  metadata: {
    name: string;
    namespace: string;
    selfLink: string;
    uid: string;
    resourceVersion: string;
    creationTimestamp: string;
    finalizers: string[];
    generation: number;
  };
  spec: {
    clusterServiceClassExternalName: string;
    clusterServicePlanExternalName: string;
    externalID: string;
    clusterServicePlanRef?: {
      name: string;
    };
    clusterServiceClassRef?: {
      name: string;
    };
  };
  status: { conditions: ICondition[] };
}

export class ServiceInstance {
  public static async create(
    releaseName: string,
    namespace: string,
    className: string,
    planName: string,
    parameters: {},
  ) {
    const { data } = await axiosWithAuth.post<IStatus>(this.getLink(namespace), {
      apiVersion: "servicecatalog.k8s.io/v1beta1",
      kind: "ServiceInstance",
      metadata: {
        name: releaseName,
      },
      spec: {
        clusterServiceClassExternalName: className,
        clusterServicePlanExternalName: planName,
        parameters,
      },
    });
    return data;
  }

  public static async get(namespace?: string, name?: string): Promise<IServiceInstance> {
    const url = this.getLink(namespace, name);
    const { data } = await axiosWithAuth.get<IServiceInstance>(url);
    return data;
  }

  public static async list(namespace?: string): Promise<IServiceInstance[]> {
    const instances = await ServiceCatalog.getItems<IServiceInstance>(
      "serviceinstances",
      namespace,
    );
    return instances;
  }

  private static getLink(namespace?: string, name?: string): string {
    return `${K8S_REVERSE_PROXY_URL}/apis/servicecatalog.k8s.io/v1beta1${
      namespace ? `/namespaces/${namespace}` : ""
    }/serviceinstances${name ? `/${name}` : ""}`;
  }
}
