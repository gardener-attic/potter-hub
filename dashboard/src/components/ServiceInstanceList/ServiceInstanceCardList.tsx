import { lowerFirst } from "lodash";
import * as React from "react";

import { IClusterServiceClass } from "shared/ClusterServiceClass";
import { IServiceInstance } from "shared/ServiceInstance";
import { CardGrid } from "../Card";
import "../Catalog/CatalogItem.css";
import ServiceInstanceCard from "./ServiceInstanceCard";
import "./ServiceInstanceCard.css";

export interface IServiceInstanceCardListProps {
  classes: IClusterServiceClass[];
  instances: IServiceInstance[];
}

const ServiceInstanceCardList: React.SFC<IServiceInstanceCardListProps> = props => {
  const { instances, classes } = props;
  return (
    <div className="ServiceInstanceCardList">
      <section>
        <CardGrid>
          {instances.length > 0 &&
            instances.map(instance => {
              const conditions = [...instance.status.conditions];
              const status = conditions.shift(); // first in list is most recent
              const svcClass = classes.find(
                potential =>
                  !!instance.spec.clusterServiceClassRef &&
                  potential.metadata.name === instance.spec.clusterServiceClassRef.name,
              );
              const broker = svcClass && svcClass.spec.clusterServiceBrokerName;
              const icon =
                svcClass &&
                svcClass.spec.externalMetadata &&
                svcClass.spec.externalMetadata.imageUrl;
              const link =
                broker &&
                `/services/brokers/${broker}/instances/ns/${instance.metadata.namespace}/${instance.metadata.name}`;

              return (
                <ServiceInstanceCard
                  key={instance.metadata.uid}
                  name={instance.metadata.name}
                  namespace={instance.metadata.namespace}
                  link={link}
                  icon={icon}
                  serviceClassName={(svcClass && svcClass.spec.externalName) || "-"}
                  servicePlanName={instance.spec.clusterServicePlanExternalName}
                  statusReason={status && lowerFirst(status.reason)}
                />
              );
            })}
        </CardGrid>
      </section>
    </div>
  );
};

export default ServiceInstanceCardList;
