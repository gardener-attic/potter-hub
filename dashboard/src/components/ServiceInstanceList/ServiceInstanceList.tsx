import * as React from "react";
import { Link } from "react-router-dom";

import { IServiceCatalogState } from "../../reducers/serviceCatalog";
import { IServiceInstance } from "../../shared/ServiceInstance";
import { IRBACRole } from "../../shared/types";
import { escapeRegExp } from "../../shared/utils";
import {
  ErrorSelector,
  MessageAlert,
  ServiceBrokersNotFoundAlert,
  ServiceCatalogNotInstalledAlert,
} from "../ErrorAlert";
import LoadingWrapper from "../LoadingWrapper";
import PageHeader from "../PageHeader";
import SearchFilter from "../SearchFilter";
import ServiceInstanceCardList from "./ServiceInstanceCardList";

export interface IServiceInstanceListProps {
  brokers: IServiceCatalogState["brokers"];
  classes: IServiceCatalogState["classes"];
  error: Error | undefined;
  filter: string;
  getBrokers: () => Promise<any>;
  getClasses: () => Promise<any>;
  getInstances: (ns: string) => Promise<any>;
  checkCatalogInstalled: () => Promise<any>;
  instances: IServiceCatalogState["instances"];
  pushSearchFilter: (filter: string) => any;
  isServiceCatalogInstalled: boolean;
  namespace: string;
}

export interface IServiceInstanceListState {
  filter: string;
}

const RequiredRBACRoles: IRBACRole[] = [
  {
    apiGroup: "servicecatalog.k8s.io",
    clusterWide: true,
    resource: "clusterservicebrokers",
    verbs: ["list"],
  },
  {
    apiGroup: "servicecatalog.k8s.io",
    clusterWide: true,
    resource: "clusterserviceclasses",
    verbs: ["list"],
  },
  {
    apiGroup: "servicecatalog.k8s.io",
    resource: "serviceinstances",
    verbs: ["list"],
  },
];

class ServiceInstanceList extends React.PureComponent<
  IServiceInstanceListProps,
  IServiceInstanceListState
> {
  public state: IServiceInstanceListState = { filter: "" };
  public async componentDidMount() {
    this.props.checkCatalogInstalled();
    this.props.getBrokers();
    this.props.getClasses();
    this.props.getInstances(this.props.namespace);
    this.setState({ filter: this.props.filter });
  }

  public componentDidUpdate(prevProps: IServiceInstanceListProps) {
    const { error, filter, getInstances, isServiceCatalogInstalled, namespace } = this.props;
    // refetch if new namespace or error removed due to location change
    if (
      isServiceCatalogInstalled &&
      (prevProps.namespace !== namespace || (error && !prevProps.error))
    ) {
      getInstances(namespace);
    }
    if (prevProps.filter !== filter) {
      this.setState({ filter });
    }
  }

  public render() {
    const {
      error,
      isServiceCatalogInstalled,
      brokers,
      instances,
      classes,
      pushSearchFilter,
    } = this.props;
    const loaded = !brokers.isFetching && !instances.isFetching && !classes.isFetching;
    let body = <span />;
    if (!isServiceCatalogInstalled) {
      body = <ServiceCatalogNotInstalledAlert />;
    } else {
      if (brokers.list.length === 0) {
        body = <ServiceBrokersNotFoundAlert />;
      } else {
        if (error) {
          body = (
            <ErrorSelector
              error={error}
              action="list"
              defaultRequiredRBACRoles={{ list: RequiredRBACRoles }}
              resource="Service Brokers, Classes and Instances"
            />
          );
        } else {
          if (instances.list.length === 0) {
            body = (
              <MessageAlert header="Provision External Services from the Kubernetes Service Catalog">
                <div>
                  <p className="margin-v-normal">
                    Kubeapps lets you browse, provision and manage external services provided by the
                    Service Brokers configured in your cluster.
                  </p>
                  <div className="padding-t-normal padding-b-normal">
                    <Link to="/services/classes">
                      <button className="button button-accent">Deploy Service Instance</button>
                    </Link>
                  </div>
                </div>
              </MessageAlert>
            );
          } else {
            body = (
              <ServiceInstanceCardList
                instances={this.filteredServiceInstances(instances.list, this.state.filter)}
                classes={classes.list}
              />
            );
          }
        }
      }
    }
    return (
      <section className="ServiceInstanceList">
        <PageHeader>
          <div className="col-8">
            <div className="row collapse-b-phone-land">
              <h1>Service Instances</h1>
              <SearchFilter
                className="margin-l-big "
                placeholder="search instances..."
                onChange={this.handleFilterQueryChange}
                value={this.state.filter}
                onSubmit={pushSearchFilter}
              />
            </div>
          </div>
          <div className="col-4 text-r align-center">
            <Link to="/services/classes">
              <button className="button button-accent">Deploy Service Instance</button>
            </Link>
          </div>
        </PageHeader>
        <MessageAlert level="warning">
          <div>
            Service Catalog integration is under heavy development. If you find an issue please
            report it{" "}
            <a target="_blank" href="https://github.com/gardener/potter-hub/issues/new/choose">
              {" "}
              here.
            </a>
          </div>
        </MessageAlert>
        <LoadingWrapper loaded={loaded}>
          <main>{body}</main>
        </LoadingWrapper>
      </section>
    );
  }

  private filteredServiceInstances(instances: IServiceInstance[], filter: string) {
    return instances.filter(i => new RegExp(escapeRegExp(filter), "i").test(i.metadata.name));
  }

  private handleFilterQueryChange = (filter: string) => {
    this.setState({
      filter,
    });
  };
}

export default ServiceInstanceList;
