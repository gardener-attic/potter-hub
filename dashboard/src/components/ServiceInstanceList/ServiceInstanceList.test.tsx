import { shallow } from "enzyme";
import context from "jest-plugin-context";
import * as React from "react";

import ServiceInstanceList from ".";
import { IServiceBroker } from "../../shared/ServiceCatalog";
import { IServiceInstance } from "../../shared/ServiceInstance";
import itBehavesLike from "../../shared/specs";
import { ForbiddenError } from "../../shared/types";
import {
  ErrorSelector,
  MessageAlert,
  ServiceBrokersNotFoundAlert,
  ServiceCatalogNotInstalledAlert,
} from "../ErrorAlert";
import ServiceInstanceCardList from "./ServiceInstanceCardList";

const defaultProps = {
  brokers: { isFetching: false, list: [] },
  classes: { isFetching: false, list: [] },
  error: undefined,
  filter: "",
  getBrokers: jest.fn(),
  getClasses: jest.fn(),
  getInstances: jest.fn(),
  checkCatalogInstalled: jest.fn(),
  instances: { isFetching: false, list: [] },
  pushSearchFilter: jest.fn(),
  isServiceCatalogInstalled: true,
  namespace: "default",
};

it("renders the warning for alpha feature", () => {
  const wrapper = shallow(<ServiceInstanceList {...defaultProps} />);
  expect(
    wrapper
      .find(MessageAlert)
      .children()
      .text(),
  ).toContain("Service Catalog integration is under heavy development");
});

context("when changing props", () => {
  it("should fetch services in the new namespace", () => {
    const getInstances = jest.fn();
    const wrapper = shallow(<ServiceInstanceList {...defaultProps} getInstances={getInstances} />);
    wrapper.setProps({ namespace: "foo" });
    expect(getInstances).toHaveBeenCalledWith("foo");
  });

  it("should update the filter", () => {
    const wrapper = shallow(<ServiceInstanceList {...defaultProps} />);
    wrapper.setProps({ filter: "foo" });
    expect(wrapper.state("filter")).toEqual("foo");
  });
});

context("while fetching components", () => {
  const props = { ...defaultProps, classes: { isFetching: true, list: [] } };

  itBehavesLike("aLoadingComponent", { component: ServiceInstanceList, props });

  it("matches the snapshot", () => {
    const wrapper = shallow(<ServiceInstanceList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a Application header", () => {
    const wrapper = shallow(<ServiceInstanceList {...props} />);
    expect(wrapper.find("h1").text()).toContain("Service Instances");
  });
});

context("when all the components are loaded", () => {
  it("shows a warning to install the Service Catalog if it's not installed", () => {
    const wrapper = shallow(
      <ServiceInstanceList {...defaultProps} isServiceCatalogInstalled={false} />,
    );
    expect(wrapper.find(ServiceCatalogNotInstalledAlert)).toExist();
    expect(wrapper).toMatchSnapshot();
  });

  it("shows a warning to install no service broker is installed", () => {
    const wrapper = shallow(
      <ServiceInstanceList
        {...defaultProps}
        // It should ignore errors if the required elements are not installed
        error={new Error()}
      />,
    );
    expect(wrapper.find(ServiceBrokersNotFoundAlert)).toExist();
    expect(wrapper).toMatchSnapshot();
  });

  it("shows a forbiden error if it exists", () => {
    const wrapper = shallow(
      <ServiceInstanceList
        {...defaultProps}
        brokers={{ isFetching: false, list: [{} as IServiceBroker] }}
        error={new ForbiddenError()}
      />,
    );
    expect(wrapper.find(ErrorSelector)).toExist();
    expect(wrapper).toMatchSnapshot();
  });

  it("shows information of how to deploy a service instance if there are none of them", () => {
    const wrapper = shallow(
      <ServiceInstanceList
        {...defaultProps}
        brokers={{ isFetching: false, list: [{} as IServiceBroker] }}
      />,
    );
    const message = wrapper.find(MessageAlert).filterWhere(m => {
      const header = m.prop("header");
      return (
        !!header && header === "Provision External Services from the Kubernetes Service Catalog"
      );
    });
    expect(message).toExist();
    expect(wrapper).toMatchSnapshot();
  });

  it("shows a card list with the different instances if they exists", () => {
    const wrapper = shallow(
      <ServiceInstanceList
        {...defaultProps}
        brokers={{ isFetching: false, list: [{} as IServiceBroker] }}
        instances={{
          isFetching: false,
          list: [
            {
              metadata: {
                name: "foo",
              },
            } as IServiceInstance,
          ],
        }}
      />,
    );
    expect(wrapper.find(ServiceInstanceCardList)).toExist();
    expect(wrapper).toMatchSnapshot();
  });
});
