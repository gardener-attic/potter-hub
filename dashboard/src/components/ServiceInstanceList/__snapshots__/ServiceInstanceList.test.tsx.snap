// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when all the components are loaded shows a card list with the different instances if they exists 1`] = `
<section
  className="ServiceInstanceList"
>
  <PageHeader>
    <div
      className="col-8"
    >
      <div
        className="row collapse-b-phone-land"
      >
        <h1>
          Service Instances
        </h1>
        <SearchFilter
          className="margin-l-big "
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search instances..."
          value=""
        />
      </div>
    </div>
    <div
      className="col-4 text-r align-center"
    >
      <Link
        to="/services/classes"
      >
        <button
          className="button button-accent"
        >
          Deploy Service Instance
        </button>
      </Link>
    </div>
  </PageHeader>
  <MessageAlertPage
    level="warning"
  >
    <div>
      Service Catalog integration is under heavy development. If you find an issue please report it
       
      <a
        href="https://github.wdf.sap.corp/kubernetes/hub/issues/new/choose"
        target="_blank"
      >
         
        here.
      </a>
    </div>
  </MessageAlertPage>
  <LoadingWrapper
    loaded={true}
    type={0}
  >
    <main>
      <ServiceInstanceCardList
        classes={Array []}
        instances={
          Array [
            Object {
              "metadata": Object {
                "name": "foo",
              },
            },
          ]
        }
      />
    </main>
  </LoadingWrapper>
</section>
`;

exports[`when all the components are loaded shows a forbiden error if it exists 1`] = `
<section
  className="ServiceInstanceList"
>
  <PageHeader>
    <div
      className="col-8"
    >
      <div
        className="row collapse-b-phone-land"
      >
        <h1>
          Service Instances
        </h1>
        <SearchFilter
          className="margin-l-big "
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search instances..."
          value=""
        />
      </div>
    </div>
    <div
      className="col-4 text-r align-center"
    >
      <Link
        to="/services/classes"
      >
        <button
          className="button button-accent"
        >
          Deploy Service Instance
        </button>
      </Link>
    </div>
  </PageHeader>
  <MessageAlertPage
    level="warning"
  >
    <div>
      Service Catalog integration is under heavy development. If you find an issue please report it
       
      <a
        href="https://github.wdf.sap.corp/kubernetes/hub/issues/new/choose"
        target="_blank"
      >
         
        here.
      </a>
    </div>
  </MessageAlertPage>
  <LoadingWrapper
    loaded={true}
    type={0}
  >
    <main>
      <ErrorSelector
        action="list"
        defaultRequiredRBACRoles={
          Object {
            "list": Array [
              Object {
                "apiGroup": "servicecatalog.k8s.io",
                "clusterWide": true,
                "resource": "clusterservicebrokers",
                "verbs": Array [
                  "list",
                ],
              },
              Object {
                "apiGroup": "servicecatalog.k8s.io",
                "clusterWide": true,
                "resource": "clusterserviceclasses",
                "verbs": Array [
                  "list",
                ],
              },
              Object {
                "apiGroup": "servicecatalog.k8s.io",
                "resource": "serviceinstances",
                "verbs": Array [
                  "list",
                ],
              },
            ],
          }
        }
        error={[Error]}
        resource="Service Brokers, Classes and Instances"
      />
    </main>
  </LoadingWrapper>
</section>
`;

exports[`when all the components are loaded shows a warning to install no service broker is installed 1`] = `
<section
  className="ServiceInstanceList"
>
  <PageHeader>
    <div
      className="col-8"
    >
      <div
        className="row collapse-b-phone-land"
      >
        <h1>
          Service Instances
        </h1>
        <SearchFilter
          className="margin-l-big "
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search instances..."
          value=""
        />
      </div>
    </div>
    <div
      className="col-4 text-r align-center"
    >
      <Link
        to="/services/classes"
      >
        <button
          className="button button-accent"
        >
          Deploy Service Instance
        </button>
      </Link>
    </div>
  </PageHeader>
  <MessageAlertPage
    level="warning"
  >
    <div>
      Service Catalog integration is under heavy development. If you find an issue please report it
       
      <a
        href="https://github.wdf.sap.corp/kubernetes/hub/issues/new/choose"
        target="_blank"
      >
         
        here.
      </a>
    </div>
  </MessageAlertPage>
  <LoadingWrapper
    loaded={true}
    type={0}
  >
    <main>
      <ServiceBrokersNotFoundAlert />
    </main>
  </LoadingWrapper>
</section>
`;

exports[`when all the components are loaded shows a warning to install the Service Catalog if it's not installed 1`] = `
<section
  className="ServiceInstanceList"
>
  <PageHeader>
    <div
      className="col-8"
    >
      <div
        className="row collapse-b-phone-land"
      >
        <h1>
          Service Instances
        </h1>
        <SearchFilter
          className="margin-l-big "
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search instances..."
          value=""
        />
      </div>
    </div>
    <div
      className="col-4 text-r align-center"
    >
      <Link
        to="/services/classes"
      >
        <button
          className="button button-accent"
        >
          Deploy Service Instance
        </button>
      </Link>
    </div>
  </PageHeader>
  <MessageAlertPage
    level="warning"
  >
    <div>
      Service Catalog integration is under heavy development. If you find an issue please report it
       
      <a
        href="https://github.wdf.sap.corp/kubernetes/hub/issues/new/choose"
        target="_blank"
      >
         
        here.
      </a>
    </div>
  </MessageAlertPage>
  <LoadingWrapper
    loaded={true}
    type={0}
  >
    <main>
      <ServiceCatalogNotInstalledAlert />
    </main>
  </LoadingWrapper>
</section>
`;

exports[`when all the components are loaded shows information of how to deploy a service instance if there are none of them 1`] = `
<section
  className="ServiceInstanceList"
>
  <PageHeader>
    <div
      className="col-8"
    >
      <div
        className="row collapse-b-phone-land"
      >
        <h1>
          Service Instances
        </h1>
        <SearchFilter
          className="margin-l-big "
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search instances..."
          value=""
        />
      </div>
    </div>
    <div
      className="col-4 text-r align-center"
    >
      <Link
        to="/services/classes"
      >
        <button
          className="button button-accent"
        >
          Deploy Service Instance
        </button>
      </Link>
    </div>
  </PageHeader>
  <MessageAlertPage
    level="warning"
  >
    <div>
      Service Catalog integration is under heavy development. If you find an issue please report it
       
      <a
        href="https://github.wdf.sap.corp/kubernetes/hub/issues/new/choose"
        target="_blank"
      >
         
        here.
      </a>
    </div>
  </MessageAlertPage>
  <LoadingWrapper
    loaded={true}
    type={0}
  >
    <main>
      <MessageAlertPage
        header="Provision External Services from the Kubernetes Service Catalog"
      >
        <div>
          <p
            className="margin-v-normal"
          >
            Kubeapps lets you browse, provision and manage external services provided by the Service Brokers configured in your cluster.
          </p>
          <div
            className="padding-t-normal padding-b-normal"
          >
            <Link
              to="/services/classes"
            >
              <button
                className="button button-accent"
              >
                Deploy Service Instance
              </button>
            </Link>
          </div>
        </div>
      </MessageAlertPage>
    </main>
  </LoadingWrapper>
</section>
`;

exports[`while fetching components loading spinner matches the snapshot 1`] = `
<section
  className="ServiceInstanceList"
>
  <PageHeader>
    <div
      className="col-8"
    >
      <div
        className="row collapse-b-phone-land"
      >
        <h1>
          Service Instances
        </h1>
        <SearchFilter
          className="margin-l-big "
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search instances..."
          value=""
        />
      </div>
    </div>
    <div
      className="col-4 text-r align-center"
    >
      <Link
        to="/services/classes"
      >
        <button
          className="button button-accent"
        >
          Deploy Service Instance
        </button>
      </Link>
    </div>
  </PageHeader>
  <MessageAlertPage
    level="warning"
  >
    <div>
      Service Catalog integration is under heavy development. If you find an issue please report it
       
      <a
        href="https://github.wdf.sap.corp/kubernetes/hub/issues/new/choose"
        target="_blank"
      >
         
        here.
      </a>
    </div>
  </MessageAlertPage>
  <LoadingWrapper
    loaded={false}
    type={0}
  >
    <main>
      <ServiceBrokersNotFoundAlert />
    </main>
  </LoadingWrapper>
</section>
`;

exports[`while fetching components matches the snapshot 1`] = `
<section
  className="ServiceInstanceList"
>
  <PageHeader>
    <div
      className="col-8"
    >
      <div
        className="row collapse-b-phone-land"
      >
        <h1>
          Service Instances
        </h1>
        <SearchFilter
          className="margin-l-big "
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search instances..."
          value=""
        />
      </div>
    </div>
    <div
      className="col-4 text-r align-center"
    >
      <Link
        to="/services/classes"
      >
        <button
          className="button button-accent"
        >
          Deploy Service Instance
        </button>
      </Link>
    </div>
  </PageHeader>
  <MessageAlertPage
    level="warning"
  >
    <div>
      Service Catalog integration is under heavy development. If you find an issue please report it
       
      <a
        href="https://github.wdf.sap.corp/kubernetes/hub/issues/new/choose"
        target="_blank"
      >
         
        here.
      </a>
    </div>
  </MessageAlertPage>
  <LoadingWrapper
    loaded={false}
    type={0}
  >
    <main>
      <ServiceBrokersNotFoundAlert />
    </main>
  </LoadingWrapper>
</section>
`;
